parser/src/main/java/log/parser/controller/MatchController.java

```diff
@@ -0,0 +1,31 @@
+package log.parser.controller;
+
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.NotNull;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import log.parser.service.MatchService;
+
+@RestController
+@RequestMapping("api/match")
+public class MatchController {
+
+	private final MatchService matchService;
+	
+	@Autowired
+	public MatchController(MatchService matchService) {
+		this.matchService = matchService;
+	}
+	
+	@PostMapping(consumes="text/plain")
+	public ResponseEntity<Long> ingestMatch(@RequestBody @NotNull @NotBlank String payload) {
```

Comment:
Suppose you are inserting users. The API design convention dictates that you should reserve the use of the standard endpoints to their corresponding actions.

- GET /users = get all
- POST /users = create
- GET /users/$id = read
- PUT or PATCH /users/$id = update
- DELETE /users/$id = delete

Because of this, I would caution against using POST /api/matches for handling file uploads. I would rather use something like POST /api/matches/upload for this purpose. Also, if the endpoint is going to create, update, delete a resource, I expect the resource to be returned as a response payload in a 201/200 response. 

As REST api design does not have any standards but only conventions, I prefer to consult the JSONAPI spec for guidance. https://jsonapi.org/format/